fcrypt/fcrypt.go

```diff
@@ -125,30 +134,41 @@ type certificateChainInfo struct {
 
 // New create context for crypto operations.
 func New(
-	provider CertificateProvider, cryptocontext *cryptutils.CryptoContext,
+	provider CertificateProvider, cryptocontext *cryptutils.CryptoContext, serviceDiscoveryURL string,
 ) (handler *CryptoHandler, err error) {
-	handler = &CryptoHandler{certProvider: provider, cryptoContext: cryptocontext}
+	handler = &CryptoHandler{
+		certProvider:        provider,
+		cryptoContext:       cryptocontext,
+		serviceDiscoveryURL: serviceDiscoveryURL,
+	}
 
 	return handler, nil
 }
 
-// GetOrganization returns online certificate origanizarion names.
-func (handler *CryptoHandler) GetOrganization() (names []string, err error) {
-	certURLStr, _, err := handler.certProvider.GetCertificate(onlineCertificate, nil, "")
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
+// GetServiceDiscoveryURLs returns service discovery URLs.
+func (handler *CryptoHandler) GetServiceDiscoveryURLs() (serviceDiscoveryURLs []string) {
+	defer func() {
+		if len(serviceDiscoveryURLs) == 0 {
+			log.Warning("Service discovery URL can't be found in certificate and will be used from config")
+
+			serviceDiscoveryURLs = append(serviceDiscoveryURLs, handler.serviceDiscoveryURL)
+		}
+	}()
+
+	certs, err := handler.getOnlineCert()
+	if err != nil || len(certs) == 0 {
+		return nil
 	}
 
-	certs, err := handler.cryptoContext.LoadCertificateByURL(certURLStr)
-	if err != nil {
-		return nil, aoserrors.Wrap(err)
+	if serviceDiscoveryURLs, err := handler.getServiceDiscoveryFromExtensions(certs[0]); err == nil {
```

Comment:
You should not mask serviceDiscoveryURLs here:
```go
if serviceDiscoveryURLs, err = handler.getServiceDiscoveryFromExtensions(certs[0])
```

Reply:
Done